### ISSUE TITLE
MLflow - MlflowException: Invalid artifact path: '/Shared/YOLOv8'

### ISSUE THREAD

**Issue by robmarkcole** [ultralytics/ultralytics#2236]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

On completion of training with MLflow logger enabled, the following exception is raised:

```
MlflowException: Invalid artifact path: '[/Shared/YOLOv8](https://file+.vscode-resource.vscode-cdn.net/Shared/YOLOv8)'. 

Names may be treated as files in certain cases, and must not resolve to other names when treated as such. 

This name would resolve to '[/Shared/YOLOv8](https://file+.vscode-resource.vscode-cdn.net/Shared/YOLOv8)
```

This arrises as logging artefacts attempts to write to '/Shared/YOLOv8', set [below](https://github.com/ultralytics/ultralytics/blob/efc941aa8131eac680872edf202b8c23be176abd/ultralytics/yolo/utils/callbacks/mlflow.py#L29)
```
experiment_name = trainer.args.project or '/Shared/YOLOv8'
```

Suggest using an alternative to  '/Shared/YOLOv8'

### Environment

Any

### Minimal Reproducible Example

It is necessary to set a valid MLflow instance url and run `model.train`

```
os.environ["MLFLOW_TRACKING_URI"] = "https://dev-mlflow.bla.com/"
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by robmarkcole** [comment]
Duplicates https://github.com/ultralytics/ultralytics/issues/1891 closing

**Comment by glenn-jocher** [comment]
@robmarkcole thank you for bringing this to our attention. This issue arises as the logging artifacts attempt to write to '/Shared/YOLOv8', which results in the exception you have provided. To resolve this issue, I suggest using a different name instead of '/Shared/YOLOv8', such as '/MyProject/YOLOv8'. Please let us know if this resolves the issue for you. Thank you for using YOLOv8.

**Comment by robmarkcole** [comment]
For future ref, the solution is to pass a `project` to `model.train`

**Comment by glenn-jocher** [comment]
@robmarkcole thank you for sharing the solution to the issue you've encountered. Providing a `project` parameter when calling `model.train` will resolve the issue and allow logging to proceed without throwing an exception. Please feel free to reach out if you have any further questions or concerns. Thank you for using YOLOv8!

**Comment by Hir98** [comment]
@glenn-jocher  I am facing same issue like
MlflowException: Invalid artifact path: '/Shared/YOLOv8'. Names may be treated as files in certain cases, and must not resolve to other names when treated as such. This name would resolve to '/Shared/YOLOv8'

How to resolve this issue ??

Can you please help me?

**Comment by robmarkcole** [comment]
@Hir98 Set the env var, the solution is to pass a project to model.train:

```
DEVICE = "mps"
PROJECT = "my_proj"
NAME = "yolov8n-run1"

model = YOLO('yolov8n.yaml')

# Train the model
model.train(
    data=f'{dataset.location}/data.yaml', 
    epochs=1, 
    imgsz=640,
    plots=True, 
    device=DEVICE, 
    project=PROJECT, 
    name=NAME, 
    verbose=True, 
    )
```

**Comment by glenn-jocher** [comment]
Hi @robmarkcole, thanks for reaching out. To resolve this issue, you should set a specific project name when you call the `model.train` method. Instead of using the '/Shared/YOLOv8' directory, which seems to be causing the exception, specify your own project directory. 

In your training script, when you call `model.train`, you can add a `project='your_project_name'` parameter. This will designate a specific project directory for logging the artifacts, preventing the error you're seeing.

Replace `'your_project_name'` with your preferred project directory name. This should solve your problem. Let us know if you have more questions.

**Comment by Hir98** [comment]
@glenn-jocher @robmarkcole 
Through above solution i am able to logs all data in colab runtime
but i want to logs or store all data in mlflow server how can i do??

Here is sample code.
`from mlflow.protos.service_pb2 import PROJECT
import mlflow

# load the configuration file
with open(r"/content/drive/MyDrive/YOLOv8/params.yaml") as f:
    params = yaml.safe_load(f)


mlflow.end_run()

# set the tracking uri
mlflow.set_tracking_uri(MLFLOW_TRACKING_URI)

# start mlflow experiment
with mlflow.start_run(run_name=params['name']):
  # load a pre-trained model
  model = YOLO(params['model_type'])
  PROJECT = "my_proj"

  # # train
  model.train(
            data='/content/drive/MyDrive/YOLOv8/dataset.yaml',
            imgsz=params['imgsz'],
            batch=params['batch'],
            epochs=params['epochs'],
            optimizer=params['optimizer'],
            lr0=params['lr0'],
            seed=params['seed'],
            pretrained=params['pretrained'],    project=PROJECT, 
)

  # log params with mlflow
  mlflow.log_param('model_type', params['model_type'])
  mlflow.log_param('epochs',params['epochs'])
  mlflow.log_param('optimizer', params['optimizer'])
  mlflow.log_param('learning_rate', params['lr0'])
`
